profiling/src/profile/internal/sample.rs

```diff
@@ -1,16 +1,17 @@
 // Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 // This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
 
-use super::{Label, StackTraceId};
+use super::{LabelId, LabelSetId, StackTraceId};
 use std::hash::Hash;
 #[derive(Eq, PartialEq, Hash)]
 pub struct Sample {
     /// label includes additional context for this sample. It can include
     /// things like a thread id, allocation size, etc
-    pub labels: Vec<Label>,
+    pub labels: LabelSetId,
 
-    /// Offset into `labels` for the label with key == "local root span id".
-    pub local_root_span_id_label_offset: Option<usize>,
+    /// This label is stored separately, to make it easy to add the endpoint
+    /// label during serialization.
+    pub local_root_span_id_label: Option<LabelId>,
```

Comment:
I had left this comment on another part of the code that was since changed but wanted to ask -- have you considered moving the `local_root_span_id_label` inside the `LabelSet`?

Since every sample that shares the same `LabelSet` will have the same `local_root_span_id_label` anyway, it seems the concept could live closer to the labels, rather than closer to the sample.

(Not blocking for this PR at all!)

Reply:
Done